torch/_inductor/fx_passes/post_grad.py

```diff
@@ -660,6 +661,97 @@ def lazy_init():
         extra_check=prepare_softmax_extra_check,
     )
 
+    register_addmm_activation_fusion()
+
+
+@functools.cache
+def register_addmm_activation_fusion():
+    shapes = [(5,), (3, 4), (4, 5)]
+    args_fp32 = [torch.empty(shape) for shape in shapes]
+    args_bf16 = [torch.empty(shape, dtype=torch.bfloat16) for shape in shapes]
+
+    for pattern in [addmm_relu_pattern, addmm_relu_pattern_2]:
+        name = f"{pattern.__name__}_fp32"
+        gen_register_replacement(
+            name,
+            pattern,
+            addmm_relu_replacement,
+            args_fp32,
+            trace_fn=fwd_only,
+            pass_dicts=pass_patterns[2],
+            extra_check=is_valid_addmm_activation_fusion,
+        )
+
+    for args, dtype_suffix in [(args_fp32, "fp32"), (args_bf16, "bf16")]:
+        for pattern in [addmm_gelu_pattern, addmm_gelu_pattern_2]:
+            name = f"{pattern.__name__}_{dtype_suffix}"
+            gen_register_replacement(
+                name,
+                pattern,
+                addmm_gelu_replacement,
+                args,
+                trace_fn=fwd_only,
+                pass_dicts=pass_patterns[2],
+                extra_check=is_valid_addmm_activation_fusion,
+            )
+
+
+def is_valid_addmm_activation_fusion(match):
+    if config.max_autotune_gemm:
+        return False
+    inp = match.kwargs["input"].meta["val"]
+    mat1 = match.kwargs["mat1"].meta["val"]
+    mat2 = match.kwargs["mat2"].meta["val"]
+
+    # match the dispatch logic for cuBLASLT at aten/src/ATen/native/cuda/Blas.cpp
+    if not (inp.is_cuda and inp.dim() == 1 and inp.is_contiguous()):
+        return False
+
+    if not (mat1.dim() == 2 and mat2.dim() == 2):
+        return False
+
+    if inp.size(0) != mat2.size(1):
+        return False
+
+    if inp.dtype != mat1.dtype or inp.dtype != mat2.dtype:
+        return False
+
+    output = match.output_node()
+    # do not fuse if there are pointwise ops after
+    return not all(is_pointwise_use(use) for use in output.users)
```

Comment:
I am a bit confused about the commend and the actual expression. The expression would be equivalent to "if there is a non-pointwise use, return Valid Fusion, else Invalid Fusion". Is, actually, `any(is_pointwise_use...)` is meant?